src/test/kotlin/pt/up/fe/ni/website/backend/controller/AuthControllerTest.kt

```diff
@@ -0,0 +1,185 @@
+import org.springframework.test.annotation.DirtiesContext
+import org.springframework.test.web.servlet.MockMvc
+import org.springframework.test.web.servlet.get
+import org.springframework.test.web.servlet.post
+import pt.up.fe.ni.website.backend.model.Account
+import pt.up.fe.ni.website.backend.model.CustomWebsite
+import pt.up.fe.ni.website.backend.repository.AccountRepository
+import pt.up.fe.ni.website.backend.utils.TestUtils
+import java.util.Calendar
+
+@SpringBootTest
+@AutoConfigureMockMvc
+@AutoConfigureTestDatabase
+@DirtiesContext(classMode = DirtiesContext.ClassMode.AFTER_CLASS)
+class AuthControllerTest @Autowired constructor(
+    val repository: AccountRepository,
+    val mockMvc: MockMvc,
+    val objectMapper: ObjectMapper,
+    passwordEncoder: PasswordEncoder
+) {
+    final val testPassword = "testPassword"
+
+    // TODO: Make sure to add "MEMBER" role to the account
+    val testAccount = Account(
+        "Test Account",
+        "test_account@test.com",
+        passwordEncoder.encode(testPassword),
+        "This is a test account",
+        TestUtils.createDate(2001, Calendar.JULY, 28),
+        "https://test-photo.com",
+        "https://linkedin.com",
+        "https://github.com",
+        listOf(
+            CustomWebsite("https://test-website.com", "https://test-website.com/logo.png")
+        )
+    )
+
+    @Nested
+    @DisplayName("POST /auth/new")
+    @TestInstance(TestInstance.Lifecycle.PER_CLASS)
+    inner class GetNewToken {
+        @BeforeAll
+        fun setup() {
+            repository.save(testAccount)
+        }
+
+        @Test
+        fun `should fail when email is invalid`() {
+            mockMvc.post("/auth/new") {
+                contentType = MediaType.APPLICATION_JSON
+                content = objectMapper.writeValueAsString(
+                    mapOf(
+                        "email" to "president@niaefeup.pt",
+                        "password" to testPassword
+                    )
+                )
+            }.andExpect {
+                status { isNotFound() }
+                jsonPath("$.errors[0].message") { value("account not found with email president@niaefeup.pt") }
+            }
+        }
+
+        @Test
+        fun `should fail when password is incorrect`() {
+            mockMvc.post("/auth/new") {
+                contentType = MediaType.APPLICATION_JSON
+                content = objectMapper.writeValueAsString(LoginDto(testAccount.email, "wrong_password"))
+            }.andExpect {
+                status { isUnprocessableEntity() }
+                jsonPath("$.errors[0].message") { value("invalid credentials") }
+            }
+        }
+
+        @Test
+        fun `should return access and refresh tokens`() {
+            mockMvc.post("/auth/new") {
+                contentType = MediaType.APPLICATION_JSON
+                content = objectMapper.writeValueAsString(LoginDto(testAccount.email, testPassword))
```

Comment:
We should use the same method here and in the test above. I personally like the first one more because it doesn't depend on the Dto implementation but it can also be more verbose

Reply:
I'll convert the first to this one since I prefer the less verbose approach